content/Snakemake_intro.md

```diff
@@ -12,27 +26,471 @@ Example usecase from GUI -> multiple scripts run manually -> bash script to run
 
 ## Workflow tools
 
+- Many languages to define a workflow
+- "Framework" for executing a workflow
+- Heavily used in bioinformatics
+
 reproducible description of workflow, "smart"
 
 ## Snakemake
 
-Why snakemake over other tools?
+Why Snakemake?
+
+- Available as Python package (pip, conda), but not python specific
+- Python based syntax (easy read)
+- Possible to extend with embedded Python code, but not necessary 
+- Snakemake workflow catalog
+
+### Installation
+
+To install Snakemake, we can use Python Package Installer (`pip`). First, ensure you have Python installed on your system.
+
+```bash
+pip install snakemake
+```
+
+To verify the installation, run:
+```bash
+snakemake -v
+```
+
+You should be able to see the version of Snakemake installed on your system.
 
 ### Snakefile
 
+We can define Snakemake workflows using files called `Snakefile`s. Workflow are defined in terms of rules. Each rule describes input files, outputs and a command or script to be run. Let's have a look at a simple, one step workflow, that will copy the content of one file into another.
+
+```bash
+rule copy:
+    input:
+	    "hello.txt"
+    output: 
+	    "hello_copy.txt"
+    shell:
+	    "cp {input} {output}"
+```
+
+This `Snakefile` defines one input file `hello.txt`, one output file `hello_copy.txt` and a shell command `cp {input} {output}`. The shell command is formatted automatically by Snakemake. The `input` and `output` are substituted using the values provided in the previous section.
+
+Before running, copy the content of the `Snakefile` into the file called `Snakefile` (without an extension) and create `hello.txt` file with some content:
+```bash
+echo "Hello Snakemake!" > hello.txt
+```
+
+To run the `Snakefile`, we run the following command:
+```bash
+snakemake --cores 1
+```
+
+Snakemake executes the `Snakefile` with the one rule in it. The content is copied! Note two things: the `--cores` flag and that we did not specified the `Snakefile`. By default, Snakemake will search for a `Snakefile` in the current directory. We can specify which `Snakefile` to run using the `-s` flag. For example:
+```bash
+snakemake --cores 1 -s Snakefile_2
+```
+
+will run the `Snakefile_2`. Thanks to that, now we can also how multiple `Snakefile`s in one directory. The `--cores` flag tells Snakemake how many resources (cores) from our computer Snakemake can use to execute the workflow. It is mandatory flag, so one cannot omit it.
```

Comment:
```suggestion
will run the `Snakefile_2`. Thanks to that, now we can also have multiple `Snakefile`s in one directory. The `--cores` flag tells Snakemake how many resources (cores) from our computer Snakemake can use to execute the workflow. It is a mandatory flag, so one cannot omit it. You may however also use `--jobs` instead of `--cores` which does not make a difference on a local computer.
```